DEVELOPMENT.md

```diff
@@ -1,60 +1,172 @@
-  - `obj`: python builtin types
-  - `stdlib`: Standard library parts implemented in rust.
-- `src`: using the other subcrates to bring rustpython to life.
-- `docs`: documentation (work in progress)
-- `py_code_object`: CPython bytecode to rustpython bytecode converter (work in
-  progress)
-- `wasm`: Binary crate and resources for WebAssembly build
-- `tests`: integration test snippets
+The contents of the Development Guide include:
+
+- [Setting up a development environment](#setting-up-a-development-environment)
+- [Code style](#code-style)
+- [Testing](#testing)
+- [Profiling](#profiling)
+- [Code organization](#code-organization)
+- [Understanding internals](#understanding-internals)
+- [Questions](#questions)
+
+## Setting up a development environment
+
+RustPython requires the following:
+
+- Rust 1.36 or higher
+    - To check Rust version: `rustc --version` 
+    - If you have `rustup` on your system, enter to update to the latest
+      stable version: `rustup update stable`
+    - If you do not have Rust installed, use [rustup](https://rustup.rs/) to
+      do so.
+- CPython version 3.7.4 or higher
+    - CPython can be installed by your operating system's package manager,
+      from the [Python website](https://www.python.org/downloads/), or
+      using a third-party distribution, such as 
+      [Anaconda](https://www.anaconda.com/distribution/).
+- [Optional] The Python package, `pytest`, is used for testing Python code
+  snippets. To install, enter `python3 -m pip install pytest`.
 
 ## Code style
 
-The code style used is the default
+The Rust code style used is the default
 [rustfmt](https://github.com/rust-lang/rustfmt) codestyle. Please format your
 code accordingly. We also use [clippy](https://github.com/rust-lang/rust-clippy)
-to detect rust code issues.
+to detect rust code issues. 
+
+Python code should follow the 
+[PEP 8](https://www.python.org/dev/peps/pep-0008/) style. We also use
+[flake8](http://flake8.pycqa.org/en/latest/) to check Python code style.
 
 ## Testing
 
-To test rustpython, there is a collection of python snippets located in the
-`tests/snippets` directory. To run those tests do the following:
+To test RustPython's functionality, a collection of Python snippets are located
+in the `tests/snippets` directory and can be run using `pytest`:
 
 ```shell
 $ cd tests
 $ pytest -v
 ```
 
-There also are some unit tests, you can run those with cargo:
+Rust unit tests can be run with `cargo`:
 
 ```shell
 $ cargo test --all
 ```
 
 ## Profiling
 
-To profile rustpython, simply build in release mode with the `flame-it` feature.
-This will generate a file `flamescope.json`, which you can then view at
+To profile RustPython, build it in `release` mode with the `flame-it` feature.
+This will generate a file `flamescope.json`, which can be viewed at
 https://speedscope.app.
 
-```sh
+```shell
 $ cargo run --release --features flame-it script.py
+```
+
+To display the default output json file:
+
+```shell
```

Comment:
The `cat flamescope.json` was mostly just to show the file it generated and that it was in json format, it's not very useful to just look at the generated json.